smart_open/tests/test_smart_open.py

```diff
@@ -1441,7 +1441,9 @@ def gzip_compress(data, filename=None):
     buf = io.BytesIO()
     buf.name = filename
     with mock.patch('time.time', _MOCK_TIME):
-        gzip.GzipFile(fileobj=buf, mode='w').write(data)
+        gz = gzip.GzipFile(fileobj=buf, mode='w')
```

Comment:
Doesn't GzipFile support the `with` context manager? That would be the preferred Python idiom, rather than calling `close()` manually.

Reply:
fixed